include/onnxruntime/core/framework/allocator.h

```diff
@@ -150,7 +156,8 @@ class IAllocator {
   static IAllocatorUniquePtr<T> MakeUniquePtr(std::shared_ptr<IAllocator> allocator, size_t count_or_bytes,
                                               bool use_reserve = false,
                                               Stream* stream = nullptr, WaitNotificationFn wait_fn = nullptr) {
-    if (allocator == nullptr) return nullptr;
+    assert(allocator != nullptr);
```

Comment:
nit: ORT_ENFORCE? I think an exception is preferable to a segfault in a release build.

Reply:
I started with ORT_ENFORCE and changed it as passing a null allocator should never happen, and must not currently as code using this isn't in general checking for a nullptr return.  

The ORT_ENFORCE adds extra binary size as this is a templatized function.

Reply:
I'll do some tests to try and minimize the binary size impact. Can try splitting out the validation here and for the CalcMemSizeForArray call so those aren't templatized on the allocation type.